test/unit/lib/FaviconFeed.test.js

```diff
@@ -258,35 +103,37 @@ describe("FaviconFeed", () => {
       assert.calledOnce(global.Services.tm.idleDispatchToMainThread);
     });
     it("should issue fetchIconFromRedirects twice on two different urls", async () => {
+      feed.getSite = sandbox.stub().returns(Promise.resolve(null));
       sandbox.spy(global.Services.tm, "idleDispatchToMainThread");
 
       await feed.fetchIcon("https://example.com");
       await feed.fetchIcon("https://another.example.com");
 
       assert.calledTwice(global.Services.tm.idleDispatchToMainThread);
     });
-    it("should cause sites to initialize with fetched sites if no sites", async () => {
-      delete feed._sitesByDomain;
-
-      await feed.fetchIcon(url);
+  });
 
-      assert.containsAllKeys(feed._sitesByDomain, ["facebook.com", "gmail.com", "mail.google.com"]);
+  describe("#getSite", () => {
+    it("should return site data if RemoteSettings has an entry for the domain", async () => {
+      const get = () => Promise.resolve([{domain: "example.com", image_url: "foo.img"}]);
+      feed._tippyTop = {get};
+      const site = await feed.getSite("example.com");
+      assert.equal(site.domain, "example.com");
+    });
+    it("should return null if RemoteSettings doesn't have an entry for the domain", async () => {
+      const get = () => Promise.resolve([]);
+      feed._tippyTop = {get};
+      const site = await feed.getSite("example.com");
+      assert.isNull(site);
+    });
+    it("should lazy init _tippyTop", async () => {
+      assert.isUndefined(feed._tippyTop);
+      await feed.getSite("example.com");
+      assert.ok(feed._tippyTop);
     });
```

Comment:
An alternative to this would be to store records in the local db:
```js
const collection = await RemoteSettings("tippytop").openCollection();
await collection.create({
  domain,
  image_url: `${url}/icon.png`
});
```

But this may require to setup IndexedDB in your tests suite, which may not be obvious